src/resources/Account.js

```diff
@@ -25,144 +25,110 @@ export default class Account extends Resource {
     })
   }
 
-  login (email, password = '') {
-    return new Promise(async (resolve, reject) => {
-      try {
-        const salt = await this.retrieveSalt(email) || ''
-        const hash = await bcrypt.hash(password, salt)
-        const result = await this.validateCredentials(email, hash)
+  async login (email, password = '') {
```

Comment:
The thing is, the outcome of the following two pieces of code is identical (ofc), I tested this with the following:

```js
test()
  .then(console.log.bind(console))
  .catch(console.error.bind(console))
```

**Version 1:**
```js
const test = async () => {
  try {
    throw new Error('Test error')
  } catch (e) {
    throw e
  }
}
```

**Version 2:**
```js
const test = async () => {
  throw new Error('Test error')
}
```

If the only reason to do this is to standardize error handling I suggest the following:

- Make a error handler helper, something like:
```js
export default async (callback) => {
  try {
    return await callback()
  } catch (error) {
    throw error
  }
}
```
- When we want to add error tracking we only have to add it in one place.

Now in the SDK functions you do the following:

```js
import errorHandler from '../helpers/error-handler'

// ...
login (email, password) {
  return errorHandler(async () => {
    // ...
  })
}

// ...
```

(Or maybe we can even automatically apply this error handler wrapper to all functions of a resource class)